### PR TITLE
DBI 3rd order GCI

### PR DESCRIPTION
**Pull request by wrightjandrew**
Added the third order GCI formula to the double blacket flow class. We probably need to change the name in the class :)

image

### CODE REVIEW

File: src/qibo/models/dbi/double_bracket.py [hunk 1]
```diff
@@ -24,6 +24,7 @@ class DoubleBracketGeneratorType(Enum):
     group_commutator = auto()
     """Use group commutator approximation"""
     # TODO: add double commutator (does it converge?)
```
Comment (marekgluza) [inline src/qibo/models/dbi/double_bracket.py:26]:
```suggestion
    """This implements the approximation 
  @wrightjandrew  --- please fill in ---- $$\approx e^{s^2[A,B]+O(s^4)$$
    which is equation (8) in [https://arxiv.org/abs/2111.12177].
    More specifically `step` is taken as $\sqrt{t}$ of the desired duration so we will be approximating as above $e^{sW}$."""
```

**Review by marekgluza — CHANGES_REQUESTED** [review]
Hi @wrightjandrew great, thanks so much! Can you please brush over the suggested doc string?

Please copy paste the final one into the PR main message around your plot.

File: src/qibo/models/dbi/double_bracket.py [hunk 2]
```diff
@@ -23,7 +23,12 @@ class DoubleBracketGeneratorType(Enum):
     """Use single commutator."""
     group_commutator = auto()
     """Use group commutator approximation"""
-    # TODO: add double commutator (does it converge?)
+    group_commutator_3 = auto()
+    """Implements: $e^{\frac{\\sqrt{5}-1}{2}sH}e^{\frac{\\sqrt{5}-1}{2}sD}e^{-sH}e^{sD}e^{\frac{3-\\sqrt{5}}{2}sH}e^{sD}
```
Comment (marekgluza) [inline src/qibo/models/dbi/double_bracket.py:27]:
```suggestion
    """Implements: $e^{\frac{\\sqrt{5}-1}{2}isH}e^{\frac{\\sqrt{5}-1}{2}isD}e^{-sH}e^{isD}e^{\frac{3-\\sqrt{5}}{2}isH}e^{isD}
```

File: src/qibo/models/dbi/double_bracket.py [hunk 3]
```diff
@@ -23,7 +23,12 @@ class DoubleBracketGeneratorType(Enum):
...
+    \approx e^{s^2[H,D]} + O(s^4)$
```
[the remark this answers is not in the document]
Comment (wrightjandrew) [reply src/qibo/models/dbi/double_bracket.py:28]: I think just a minus sign

**Review by Edoardo-Pedicillo — APPROVED** [review]

**Review by andrea-pasquale — APPROVED** [review]
Thanks @wrightjandrew, just a small correction :)

**Review by marekgluza — APPROVED** [review]

**Comment by marekgluza** [comment]
@andrea-pasquale @Edoardo-Pedicillo can you merge?

**Pull request merged**